evm/v0.5/contracts/PreCoordinator.sol

```diff
@@ -233,11 +234,13 @@ contract PreCoordinator is ChainlinkClient, Ownable, ChainlinkRequestInterface,
     bytes32 saId = serviceAgreementRequests[_requestId];
     delete requests[_requestId];
     delete serviceAgreementRequests[_requestId];
+    emit ServiceAgreementRequestAnswer(saId, _data);
     if (requesters[cbRequestId].responses.push(_data) == minResponses) {
       serviceAgreements[saId].activeRequests = serviceAgreements[saId].activeRequests.sub(1);
       Requester memory req = requesters[cbRequestId];
       delete requesters[cbRequestId];
       int256 result = Median.calculate(req.responses);
+      emit ServiceAgreementRequestFulfilled(saId, cbRequestId, result);
```

Comment:
@thodges-gh To confirm this will be emitted for each incoming value. Can the saId be used to identify which node operator the value is coming from?

Reply:
`ServiceAgreementRequestAnswer` is emitted for each incoming value, `ServiceAgreementRequestFulfilled` is emitted only when the consuming contract will be pinged with the aggregate of answers.